OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorSelectionLogic.cs

```diff
@@ -124,5 +138,24 @@ void CreateCategoryPanel(MapCopyFilters copyFilter, CheckboxWidget checkbox)
 			checkbox.IsVisible = () => true;
 			checkbox.OnClick = () => copyFilters ^= copyFilter;
 		}
+
+		void HandleSelectionChanged()
+		{
+			var selectedRegion = editor.DefaultBrush.Selection.Area;
+			if (selectedRegion == null)
+				return;
+			var selectionSize = selectedRegion.BottomRight - selectedRegion.TopLeft + new CPos(1, 1);
+			var diagonalLength = Math.Round(Math.Sqrt(Math.Pow(selectionSize.X, 2) + Math.Pow(selectionSize.Y, 2)), 3);
```

Comment:
I'm wondering wether diagonal is even useful. It's kinda redundant

Reply:
If players spawns are in corners and you wanna mesure distance from spawn point to the center. Or Distance of anything from anything. I use it a lot. 
But problem is that Diagonal length doenst mach Cell length.  And I dont know how to fix this.
![Snímka obrazovky (40)](https://github.com/OpenRA/OpenRA/assets/100044015/7b2a47d0-1b3a-4161-9291-114bb367c841)

Reply:
ah I see, it can be used as a ruler. This is probably the better solution than adding an actual ruler tool